foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/generator/util/MavenUtil.java

```diff
@@ -214,6 +214,45 @@ public String getPackaging() {
             return packaging;
         }
     }
+
+    /**
+     * Reads the given POM file to determine what profile is used in data delivery pipeline
+     *
+     * @param context GenerationContext
+     * @return the profile for data delivery pipeline
+     */
+    public static String getDataDeliveryProfileName(GenerationContext context) {
+
+        File pomFile = new File(context.getProjectDirectory() + File.separator + "pom.xml");
+
+        try {
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+
+            Document doc = dBuilder.parse(pomFile);
+            doc.getDocumentElement().normalize();
+            //Get all fields with tag <module>
+            NodeList nodeList = doc.getElementsByTagName("profile");
+            //Iterate through the list and get deploy-module name
+            for (int i = 0; i < nodeList.getLength(); i++){
+                Node node = nodeList.item(i);
+
+                if (node.getNodeType() == Node.ELEMENT_NODE){
+
+                    Element element = (Element) node;
+                    String field = element.getTextContent();
+                    if(field.startsWith("data-delivery"))
+                    {
+                        return field;
```

Comment:
Q: it seems we always return the first profile name starting with `data-delivery`, is it possible that we have multiple profile name starting with `data-delivery` and can we be more specific what profile name we are looking for?

Reply:
yup I missed that, this could cause issues when there're multiple profile that starts data-delivery by looking at the profiles.json
What I am seeking for is whether pom has data-delivery-pyspark or data-delivery-pyspark.
Upon investigating profiles.json, we have only two profiles that starts with data-delivery-pyspark so Ill update to start with data-delivery-pyspark.

In some rare edge case we add data-delivery-pyspark and data-delivery-pyspark-pipeline in a pom file, that's okay. There will be many duplication in code but we have cases that supports both, so just getting first profile starting data-delivery-pyspark is good.